grav/hub.go

```diff
@@ -29,18 +33,30 @@ type hub struct {
 }
 
 type connectionHolder struct {
-	Conn         Connection
-	BelongsTo    string
-	Capabilities []string
+	Conn      Connection
+	BelongsTo string
+	Interests []string
 }
 
 func initHub(nodeUUID string, options *Options, connectFunc func() *Pod) *hub {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	withdraw := func() {
+		cancel()
+
+		// this sleep matches the sleep below so that the hub.withdraw
+		// method doesn't return until the withdraw sequence completes
+		time.Sleep(time.Second * 3)
```

Comment:
Since this needs to match the below time, perhaps it should be defined at the package level as a `const`

```suggestion
		time.Sleep(withdrawTimeout)
```